src/org/omegat/core/data/RealProject.java

```diff
@@ -1848,7 +1856,7 @@ public void addTranslation(String id, String source, String translation, boolean
                             }
                             tr.source = oneSrc;
                             tr.translation = oneTrans;
-                            data.put(sourceS, new TMXEntry(tr, true, null));
+                            data.put(new EntryKey(sourcePath, sourceS, id, "", "", oneSrc), new TMXEntry(tr, true, null));
```

Comment:
I discovered a small bug here when I was reviewing this: https://sourceforge.net/p/omegat/bugs/1104/ 
I think we should decide what to do in this case before considering this patch.

In any case, sourceS is not the correct source (and as a consequence, not the correct parameter in EntryKey) and oneSrc is not the correct last parameter ("path") in EntryKey's constructor. Here we have no "path".

What we can do:
- if we want only one entry per paragraph, then we must concatenate all sources in tr.source and all translations in tr.translation, then the call to data.put is correct
- if we want one entry per segment, then call to data.put should be part of the loop, and each time, EntryKey uses oneSrc as a source and nothing as "path"